packages/typescript-client/test/parser.test.ts

```diff
@@ -0,0 +1,98 @@
+import { describe, expect, it } from 'vitest'
+import { defaultParser, pgArrayParser } from '../src/parser'
+
+describe(`Default parser`, () => {
+  it(`should parse integers`, () => {
+    expect(defaultParser.int2(`0`)).toBe(0)
+    expect(defaultParser.int2(`-32768`)).toBe(-32768)
+    expect(defaultParser.int2(`32767`)).toBe(32767)
+
+    expect(defaultParser.int4(`0`)).toBe(0)
+    expect(defaultParser.int4(`2147483647`)).toBe(2147483647)
+    expect(defaultParser.int4(`-2147483648`)).toBe(-2147483648)
+  })
+
+  it(`should parse bigints`, () => {
+    expect(defaultParser.int8(`-9223372036854775808`)).toBe(
+      BigInt(`-9223372036854775808`)
+    )
+    expect(defaultParser.int8(`9223372036854775807`)).toBe(
+      BigInt(`9223372036854775807`)
+    )
+    expect(defaultParser.int8(`0`)).toBe(BigInt(`0`))
+  })
+
+  it(`should parse booleans`, () => {
+    expect(defaultParser.bool(`t`)).toBe(true)
+    expect(defaultParser.bool(`true`)).toBe(true)
+    expect(defaultParser.bool(`false`)).toBe(false)
+  })
+
+  it(`should parse float8`, () => {
+    expect(defaultParser.float8(`1.797e308`)).toBe(1.797e308)
+    expect(defaultParser.float8(`-1.797e+308`)).toBe(-1.797e308)
+    expect(defaultParser.float8(`0`)).toBe(0)
+  })
+
+  it(`should parse json`, () => {
+    expect(defaultParser.json(`true`)).toEqual(true)
+    expect(defaultParser.json(`5`)).toEqual(5)
+    expect(defaultParser.json(`"foo"`)).toEqual(`foo`)
+    expect(defaultParser.json(`{"a":1}`)).toEqual({ a: 1 })
+    expect(defaultParser.json(`{"a":1,"b":2}`)).toEqual({ a: 1, b: 2 })
+    expect(defaultParser.json(`[{"a":1,"b":2},{"c": [{"d": 5}]}]`)).toEqual([
+      { a: 1, b: 2 },
+      { c: [{ d: 5 }] },
+    ])
+
+    expect(defaultParser.json(`true`)).toEqual(true)
+    expect(defaultParser.json(`5`)).toEqual(5)
+    expect(defaultParser.json(`"foo"`)).toEqual(`foo`)
+    expect(defaultParser.jsonb(`{"a":1}`)).toEqual({ a: 1 })
+    expect(defaultParser.jsonb(`{"a":1,"b":2}`)).toEqual({ a: 1, b: 2 })
+    expect(defaultParser.jsonb(`[{"a":1,"b":2},{"c": [{"d": 5}]}]`)).toEqual([
+      { a: 1, b: 2 },
+      { c: [{ d: 5 }] },
+    ])
+  })
+})
+
+describe(`Postgres array parser`, () => {
+  it(`should parse arrays and their values`, () => {
+    expect(pgArrayParser(`{1,2,3,4,5}`, defaultParser.int2)).toEqual([
+      1, 2, 3, 4, 5,
+    ])
+    expect(pgArrayParser(`{1,2,3,4,5}`, defaultParser.int8)).toEqual([
+      BigInt(1),
+      BigInt(2),
+      BigInt(3),
+      BigInt(4),
+      BigInt(5),
+    ])
+    expect(pgArrayParser(`{"foo","bar"}`, (v) => v)).toEqual([`foo`, `bar`])
```

Comment:
How about an addiitonal test over real edge cases?
```suggestion
    expect(pgArrayParser(`{"foo","bar"}`, (v) => v)).toEqual([`foo`, `bar`])
    expect(pgArrayParser(`{foo,"}"}`, (v) => v)).toEqual([`foo`, `}`])
```

Reply:
I added some corner cases, you can have a look to see if it covers them all or if you want some additional ones